pkgs/test_api/lib/hooks_testing.dart

```diff
@@ -0,0 +1,156 @@
+  /// ```
+  static Future<TestCaseMonitor> run(FutureOr<void> Function() body) async {
+    final monitor = TestCaseMonitor.start(body);
+    await monitor.onDone;
+    return monitor;
+  }
+
+  /// Start [body] as a test case and return a [TestCaseMonitor] with the status
+  /// and result.
+  ///
+  /// The [state] will start as [State.pending] if queried synchronously, but it
+  /// will switch to [State.running]. After `onDone` completes the state will be
+  /// one of [State.passed], [State.skipped], or [State.failed].
+  ///
+  /// {@macro result-late-fail}
+  ///
+  /// ```dart
+  /// late void Function() completeWork;
+  /// final monitor = TestCaseMonitor.start(() {
+  ///   final outstandingWork = TestHandle.current.markPending();
+  ///   completeWork = outstandingWork.complete;
+  /// });
+  /// await pumpEventQueue();
+  /// assert(monitor.state == State.running);
+  /// completeWork();
+  /// await monitor.onDone;
+  /// assert(monitor.state == State.passed);
+  /// ```
+  static TestCaseMonitor start(FutureOr<void> Function() body) =>
+      TestCaseMonitor._(body).._start();
+
+  void _start() {
+    _liveTest.run().whenComplete(_done.complete);
+  }
+
+  /// A future that completes after this test has finished running, or has
+  /// surfaced an error.
+  Future<void> get onDone => _done.future;
+
+  /// The running and success or failure status for the test case.
+  State get state {
+    final status = _liveTest.state.status;
+    if (status == Status.pending) return State.pending;
+    if (status == Status.running) return State.running;
+    final result = _liveTest.state.result;
+    if (result == Result.skipped) return State.skipped;
+    if (result == Result.success) return State.passed;
+    // result == Result.failure || result == Result.error
+    return State.failed;
+  }
+
+  /// The errors surfaced by the test.
+  ///
+  /// A test with any errors will have a failing [result].
+  ///
+  /// {@macro result-late-fail}
+  ///
+  /// A test may have more than one error if there were unhandled asynchronous
+  /// errors surfaced after the test is done.
+  Iterable<Object> get errors => _liveTest.errors.map((a) => a.error);
+
+  /// A stream of errors surfaced by the test.
+  ///
+  /// This stream will not close, asynchronous errors may be surfaced within the
+  /// test's error zone at any point.
+  Stream<Object> get onError => _liveTest.onError.map((a) => a.error);
+}
+
+/// Returns a local [LiveTest] that runs [body].
+LiveTest _createTest(FutureOr<void> Function() body) {
+  var test = LocalTest('test', Metadata(chainStackTraces: true), body);
+  var suite = Suite(Group.root([test]), _suitePlatform, ignoreTimeouts: false);
+  return test.load(suite);
+}
+
+/// A dummy suite platform to use for testing suites.
+final _suitePlatform =
+    SuitePlatform(Runtime.vm, compiler: Runtime.vm.defaultCompiler);
+
+/// The running and success state of a test monitored by a [TestCaseMonitor].
+enum State {
```

Comment:
@grouma @jakemac53 - WDYT about creating this new abstraction to hide some of the details of `Result` and `Status`?

I don't like the idea of introducing definitions for the term `State`, but I also don't like the idea of exposing the messy details when we could have a much simpler model for users.

Reply:
> but I also don't like the idea of exposing the messy details

What are the messy details? It looks like `State` is just a merge of `Result` and `Status`.

Reply:
There is more behavior to document if we expose the `Result` and `Status`.
- We'd have to talk about how `Result` doesn't have meaning until `Status` is `Status.complete`.
- We'd have to talk about how users of these APIs will never see a `Status.pending`.
- We'd have to talk about, and carry forward, the "error" vs "failure" distinction which we'd like to move away from. https://github.com/dart-lang/test/issues/1465 

We have an existing `State` class which holds a `Result` and `Status`, but I think it's friendlier to model is as the single enum instead of exposing both.

Overall, it does feel weird to have 2 enums which are not orthogonal. When the value of one field is interesting, the other field can only have 1 value, and vice versa.

Reply:
Ah yes. Your proposed changes definitely help and LGTM.